plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeRegisterTableProcedureTest.java

```diff
@@ -177,6 +177,29 @@ public void testRegisterTableWithDifferentTableName()
         assertUpdate(format("DROP TABLE %s", tableNameNew));
     }
 
+    @Test
+    public void testRegisterTableWithTrailingSpaceInLocation()
+    {
+        String tableName = "test_create_table_with_trailing_space_" + randomNameSuffix();
+        String tableLocationWithSpace = tableName + " ";
+
+        assertQuerySucceeds(format("CREATE TABLE %s WITH (location = '%s') AS SELECT 1 as a, 'INDIA' as b, true as c", tableName, tableLocationWithSpace));
+        assertQuery("SELECT * FROM " + tableName, "VALUES (1, 'INDIA', true)");
+
+        String tableLocation = getTableLocation(tableName);
+        assertThat(tableLocation).isEqualTo(tableLocationWithSpace);
+
+        String registeredTableName = "test_register_table_with_trailing_space_" + randomNameSuffix();
+        assertQuerySucceeds(format("CALL system.register_table('%s', '%s', '%s')", SCHEMA, registeredTableName, tableLocation));
```

Comment:
I would replace `tableLocation` with `tableLocationWithSpace` to clarify the test purpose.